include/seqan3/range/view/translate_join.hpp

```diff
@@ -0,0 +1,390 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Sara Hetzel <sara.hetzel AT fu-berlin.de>
+ * \brief Provides seqan3::view::translate_join.
+ */
+
+#pragma once
+
+#include <vector>
+#include <stdexcept>
+
+#include <seqan3/core/type_traits/range.hpp>
+#include <seqan3/range/container/small_string.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+#include <seqan3/range/view/translate.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/ranges>
+
+namespace seqan3::detail
+{
+
+// ============================================================================
+//  view_translate_join (range definition)
+// ============================================================================
+
+/*!\brief The return type of seqan3::view::translate_join.
+ * \implements std::ranges::View
+ * \implements std::ranges::SizedRange
+ * \implements std::ranges::RandomAccessRange
+ * \tparam urng_t The type of the range being translated.
+ * \param[in] tf Translation frames to be used.
+ * \ingroup view
+ */
+template <std::ranges::View urng_t>
+//!\cond
+    requires std::ranges::SizedRange<urng_t> &&
```

Comment:
I guess those requirements are not needed for overload resolution but just for "type cheking"? If so we decided to use static asserts rather than concet. The error messages are more readable :)

Reply:
Isn't it enough that the concepts are checked via static_assert in the adaptor? This gets anyways called when constructing the range, right?

Reply:
There is a general rule in SeqAn about having static_asserts instead of requirements if they are not required for overload resolution. For the regular translate views I would argue that we might actually add other overloads and in that case the constraints for this `class` would be the same, but we would change the constraints on the adaptor (which would then allow e.g. BidirectionalRange but would forward those to `view::translate | view::join`).

TLDR: It doesn't make much of a difference. Since I don't think this particular class will get another specialisation, you can also just copy the static assert from the adaptor into the class and remove the concept constraints.